index/scorch/merge.go

```diff
@@ -59,7 +59,13 @@ OUTER:
 				// lets get started
 				err := s.planMergeAtSnapshot(ourSnapshot, mergePlannerOptions)
 				if err != nil {
-					s.fireAsyncError(fmt.Errorf("merging err: %v", err))
+					if err == ErrClosed {
+						// if the index was closed while in the process of merging,
+						// just log a message before closing
+						log.Printf("merging err: %v", err)
```

Comment:
ok, so had a long conversation with @steveyen 

1.  let's not log this, feels like it's only going to be confusing seeing an error that's not really an error
2.  but, let's add a break OUTER, this should ensure that we directly exit the goroutine, not go around the loop again and have to detect closing a second time (raising chances that somehow we end up not actually exiting this goroutine)

Reply:
Done.